File: backend-dotnet/src/Platform.Api/Contracts/ApiResponse.cs
namespace Platform.Api.Contracts;

/// <summary>
/// Matches the existing admin-frontend's expected response envelope
/// ({ success, statusCode, message, data }) to minimize frontend churn during migration.
/// </summary>
public record ApiResponse<T>(bool Success, int StatusCode, string Message, T? Data)
{
    public static ApiResponse<T> Ok(T data, string message = "OK", int statusCode = 200) =>
        new(true, statusCode, message, data);

    public static ApiResponse<T> Fail(string message, int statusCode = 400) =>
        new(false, statusCode, message, default);
}

File: android/app/src/main/java/com/porttennanttandoori/pos/data/model/ApiResponse.kt
package com.porttennanttandoori.pos.data.model

import kotlinx.serialization.Serializable

/** Mirrors the backend's { success, statusCode, message, data } envelope (see ApiResponse.cs). */
@Serializable
data class ApiResponse<T>(
    val success: Boolean,
    val statusCode: Int,
    val message: String,
    val data: T? = null,
)
